src/main/java/com/unstoppabledomains/resolution/naming/service/ENS.java

```diff
@@ -75,6 +76,37 @@ public String getOwner(String domain) throws NamingServiceException {
     return owner;
   }
 
+  @Override
+  public List<String> batchOwners(List<String> domains) throws NamingServiceException {
+    if (domains.size() > 255) {
+      throw new NamingServiceException(NSExceptionCode.MaxThreadLimit, new NSExceptionParams("m|l", "Ens#batchOwners", "255"));
+    }
+    // We want String[] instead of List<String> to maintain the same order as domains
+    String[] owners = new String[domains.size()];
```

Comment:
@JohnnyJumper  This call will create up to 255 threads which could affect performance of Android and other low-power devices. Do you think we can afford use so much threads?

Reply:
We could leave this method unimplemented for now since we don't have fast solution. As a long-term we can consider using universal Multicall smart contract which will aggregate unlimited amount of calls on one single RPC call.

https://github.com/makerdao/multicall/blob/master/src/Multicall.sol

Reply:
Looks like MultiCall contract doesn't allow us to make the calls for free. 
https://rinkeby.etherscan.io/address/0x42ad527de7d4e9d9d011ac45b31d8551f8fe9821#readContract
I found this contract when I was developing this but I stumbled upon some problems with it.
The methods available for free calls are following:
```
getCurrentTimeStamp,
getLastBlockHash,
getEthBalance,
getCurrentBlockDifficulty,
getCurrentBlockGasLimit,
getCurrentBlockCoinbase,
getBlockHash
```

It doesn't look like these methods can help us. 
Also the write methods such  `aggregate` not only take some fees but also not returning the result? 
It is possible that I used the etherscan somehow differently than intended, but I wasn't able to get the results from rinkeby test contract.

Reply:
Regarding 255 threads: I googled what is the maximum amount of threads a JVM can handle, and decided to limit the function to 255. Also, what is the use case when a mobile application needs to get the batch results from parsing .zil or .eth address? I doubt that this method is going to be used for mobile applications. 

Also, I am unsure what are the other options to make an async call in java without the threads. If each thread is very expensive what do modern applications are using instead?

Reply:
@JohnnyJumper Etherscan lies, the aggregate method doesn't make any writings in blockchain according the code and it's able to return data according to contract source code. It's just an example and we will implement iterations with multicall contract later.

Reply:
@JohnnyJumper Threads are expensive for computations, best practice is to create no more than (cpu cores amount *2)